### PR TITLE
daq: update to 2.0.7

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/daq/Portfile [hunk 1]
```diff
@@ -17,12 +17,22 @@ depends_lib         port:pcre \
...
-                    size    518013
-# There was a stealth update of the distfile - hence the next line is required
-dist_subdir         ${name}/${version}_1
-patchfiles          patch-sfbpf-Makefile.in.diff
+checksums           rmd160  c915d34c0ba6c7deb469d6969bc1e077a27cf576 \
+                    sha256  bdc4e5a24d1ea492c39ee213a63c55466a2e8114b6a9abed609927ae13a7705e \
+                    size    520287
+
+pre-configure {
+    delete ${worksrcpath}/configure
```
Comment (ryandesign) [inline net/daq/Portfile:25]: Why does the configure script need to be deleted? Doesn't running autoreconf automatically overwrite it?

File: net/daq/Portfile [hunk 2]
```diff
@@ -17,12 +17,22 @@ depends_lib         port:pcre \
...
+}
+
+# Fix intermittent parallel build failure (Moved from the patch file)
+# https://trac.macports.org/ticket/45159
+post-configure {
+    reinplace "s|sf_scanner.c: \$(srcdir)/scanner.l|sf_scanner.c: \$(srcdir)/scanner.l \${builddir}/tokdefs.h|" ${worksrcpath}/sfbpf/Makefile
```
Comment (ryandesign) [inline net/daq/Portfile:31]: Why is this changed from a patchfile to a reinplace? Patchfiles are preferred.

@roederja What is the status of getting the developers to accept this patch, which was sent to them 6 years ago?

File: net/daq/Portfile [hunk 3]
```diff
@@ -17,12 +17,22 @@ depends_lib         port:pcre \
...
+}
+
+use_autoreconf      yes
+autoreconf.args     -fvi
```
Comment (ryandesign) [inline net/daq/Portfile:35]: Why does autoreconf need to be run?

Comment (i0ntempest) [reply net/daq/Portfile:35]: If not run it will ask for aclocal (and other autoreconf related programs) during build and errors out.

File: net/daq/Portfile (hunk 2 above)
Comment (i0ntempest) [reply net/daq/Portfile:31]: Patch (if we’re patching makefile.in) needs to be done precisely after autoreconf and before configure. Is this possible?

File: net/daq/Portfile (hunk 1 above)
Comment (i0ntempest) [reply net/daq/Portfile:25]: Well this is a weird one, on my system autoreconf can’t delete it and spews out permission denied. Maybe you can try on yours and see if it also happens?

**Comment by ryandesign** [comment]
Seems the cause of all of this is a packaging error: configure.ac is newer than the files that were generated from it. For example:

```
$ ls -lT configure*
-r-xr-xr-x  1 macports  wheel  497485 Apr  8 07:23:57 2020 configure
-r--r--r--  1 macports  wheel   11296 Apr  8 07:23:58 2020 configure.ac
```

The solution is to either `touch` all the files that were generated from configure.ac so that they have newer timestamps, but there's several of them, or more simply to change configure.ac's timestamp to match configure's. I'll push a change to do that in a moment.

File: net/daq/Portfile (hunk 1 above)
Comment (ryandesign) [reply net/daq/Portfile]: Maybe due to another packaging error: none of the files have the write permission bit set.

**Comment by i0ntempest** [comment]
Thanks... TIL configure scripts check timestamps.

**Comment by ryandesign** [comment]
> TIL configure scripts check timestamps

Yeah... Makefiles too. The idea is: if you modify configure.ac, you would want configure to be regenerated from it. And if you modify Makefile.am, you would want Makefile.in to be regenerated from it. That's why it's important to preserve timestamps properly, which I think the developers didn't do when they packaged up this tarball.
